mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+        used to generate white noise. Default value is 0.0.
+    std : float
+        Standard deviation of the Gaussian distribution
+        (np.random.normal) used to generate white noise.
+        Default value is 0.2.
+
+    Returns
+    -------
+    pac_signal : ndarray, shape (n_epochs, n_times)
+        Phase amplitude coupled signal.
+    """
+
+    times = np.linspace(0, duration, sfreq * duration)
+    n_times = len(times)
+
+    if n_epochs <= 1:
+        raise RuntimeError('No. of trials too low, please provide atleast 2.')
+
+    rng = check_random_state(random_state)
+    # generate white noise
+    white_noise = rng.normal(mean, std, n_times * n_epochs)
+    white_noise = np.reshape(white_noise, (n_epochs, n_times))
+
+    if sigma == 0:
+        raise ValueError('Sigma cannot be 0. Please choose a higher value.')
+
+    # make a gaussian window
+    win = gaussian(n_epochs, sigma, sym=False)
+
+    if np.max(win) == np.min(win):
+        raise RuntimeError('win values lead to divide by 0.')
+
+    # normalize the gaussian window
+    win = (win - np.min(win)) / (np.max(win) - np.min(win))
+
+    # Construct the signal with many levels of modulation
+    pac_signal = np.zeros((n_epochs, n_times))
+
+    for sig in range(n_epochs):
+        # Generate amplitude envelope of modulated signal
+        amp_fa = (max_amp * ((1. - win[sig]) *
+                             np.sin(2 * np.pi * f_phase * times)
+                             + 1. + win[sig]) / 2.)
+        # Generate the phase amplitude coupled signal
+        pac_signal[sig] = (amp_fa * np.sin(2. * np.pi * f_amplitude * times) +
+                           max_amp * np.sin(2. * np.pi * f_phase * times) +
+                           white_noise[sig])
+
+    if len(pac_signal) != n_epochs:
+        raise RuntimeError('Data not equal to epochs.')
+
+    # mix the trials
+    order = np.argsort(rng.randn(len(pac_signal)))
+    pac_signal = pac_signal[order, :]
+
+    return pac_signal
+
+
+def cross_frequency_coupling(data, sfreq, phase_freq, n_cycles,
+                             freqs, tmin, tmax, n_jobs=1):
+    """ Compute the cross frequency coupling.
+
+    Parameters
+    ----------
+    data : array, shape (n_times,)
+        Signal time series.
+    sfreq : float
+        Sampling frequency.
+    phase_freq : float
+        The phase modulating frequency in Hz. (eg. 10.)
+    n_cycles : float | array of float
+        Number of cycles used to compute Morlet wavelets.
+    freqs : array
+        Frequency range of interest (1 x Frequencies)
+    tmin : float
+        Start time before event.
+    tmax : float
+        End time after event.
+    n_jobs : int
+        Number of jobs to run in parallel. Default 1.
+
+    Returns
+    -------
+    times : array, shape (n_times,)
+        Time points for signal plotting.
+    ampmat : ndarray (nfreq, n_samples)
+        Amplitude traces.
+    traces : ndarray, shape (n_epochs, n_times)
```

Comment:
The 'n_epochs' here should be n_freqs, it's a error. 

Yes, `traces` was adopted because it was used in the Canolty paper as well. I'm not sure if using `evoked` will make it clearer... How about simply `spectral_amplitudes`, `raw_amplitudes`... ?